models/mart/mart_speeches.sql

```diff
@@ -231,7 +227,42 @@ with
                 else null
             end as ministry_addressed
         from extract_minister_for
+    ),
+
+    agg_ministry_counts_by_topic as (
+        select
+            topic_id,
+            ministry_addressed,
+            count(*) as ministry_count,
+            rank() over (partition by topic_id order by count(*) desc) as ministry_rank
+        from extract_ministry
+        group by topic_id, ministry_addressed
+    ),
+
+    get_majority_ministry as (
+        select
+            topic_id,
+            array_agg(ministry_addressed order by ministry_count desc limit 1)[
+                offset(0)
+            ] as topic_ministry,
+            safe_divide(
+                max(ministry_count), sum(ministry_count) over (partition by topic_id)
+            ) as topic_ministry_proportion
+        from agg_ministry_counts_by_topic
+        where ministry_rank = 1
+        group by topic_id, ministry_count
```

Comment:
```suggestion
        select
            topic_id,
            ministry_addressed as topic_ministry,
            safe_divide(
                max(ministry_count), sum(ministry_count) over (partition by topic_id)
            ) as topic_ministry_proportion
        from agg_ministry_counts_by_topic
        qualify row_number() over(partition by topic_id order by ministry_count desc) = 1
```

issue: i would have implemented it differently. this is clearer vis-a-vis the array_agg field.

Reply:
There is a difference though; the query randomly selects the top ministry via `limit 1` if there is a tie. In some cases in the top ministries addressed in a topic are split 50/50.